native/src/modules/categories/components/CategoryListContent.js

```diff
@@ -0,0 +1,181 @@
+import { type StyledComponent } from 'styled-components'
+import type { NavigationStackProp } from 'react-navigation-stack'
+import type { NavigateToInternalLinkParamsType } from '../../app/createNavigateToInternalLink'
+import type { ThemeType } from '../../../../build-configs/ThemeType'
+import { RTL_LANGUAGES } from '../../i18n/constants'
+import onLinkPress from '../../common/onLinkPress'
+
+const VerticalPadding: StyledComponent<{}, {}, *> = styled.View`
+  padding: 0 20px;
+`
+
+// type inferred from 'react-native/Libraries/Utilities/Dimensions'
+type DimensionsEventType = {
+  window?: DisplayMetrics,
+  screen?: DisplayMetrics,
+  ...
+}
+
+type ContentPropsType = {|
+  content: string,
+  navigation: NavigationStackProp<*>,
+  resourceCacheUrl: string,
+  navigateToInternalLink?: NavigateToInternalLinkParamsType => void,
+  cacheDictionary: { [string]: string },
+  language: string,
+  theme: ThemeType
+|}
+
+class CategoryListContent extends React.Component<ContentPropsType, {| width: number |}> {
+  constructor () {
+    super()
+    this.state = {
+      width: Dimensions.get('window').width
+    }
+  }
+
+  onChange = (dimensionsEvent: DimensionsEventType) => {
+    const defaultWidth = 200
+    this.setState({ width: (dimensionsEvent.window?.width || dimensionsEvent.screen?.width || defaultWidth) })
+  }
+
+  componentDidMount () {
+    Dimensions.addEventListener('change', this.onChange)
+  }
+
+  componentWillUnmount () {
+    Dimensions.removeEventListener('change', this.onChange)
+  }
+
+  onLinkPress = (evt: GestureResponderEvent, url: string) => {
+    const { language, navigation, navigateToInternalLink } = this.props
+    onLinkPress(url, navigation, language, navigateToInternalLink)
+  }
+
+  alterResources = (node: HTMLNode) => {
+    const { cacheDictionary } = this.props
+    if (node.attribs) {
+      if (node.attribs.href) {
+        console.debug(`Found href: ${decodeURI(node.attribs.href)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.href)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.href} with ${newResource}`)
+          node.attribs = {
+            ...(node.attribs || {}),
+            href: newResource
+          }
+          return node
+        }
+      } else if (node.attribs.src) {
+        console.debug(`Found src: ${decodeURI(node.attribs.src)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.src)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.src} with ${newResource}`)
+          node.attribs = {
+            ...(node.attribs || {}),
+            src: newResource
+          }
+          return node
+        }
+      }
+    }
+  }
+
+  // see https://github.com/archriss/react-native-render-html/issues/286
+  renderBulletLists: RendererFunction = (htmlAttribs, children, convertedCSSStyles, passProps) => {
+    const { language, theme } = this.props
+    const style = _constructStyles({
```

Comment:
Is this an internal API? If yes then we should probably not use it.

Reply:
It is, but there is no other workaround right now to render lists with RTL layout. As @Taggotty mentioned she well create an issue to remove this once it is fixed in the library.

Reply:
If it is really foreseeable that this function gets removed in the near future then we can merge this as is. But if not then we should improve this and not beautify the html rendering. Also the issue does not mention anything about the bullet points.

I dont think this feature is that important that it justifies to use internal API of an alpha npm module.

Reply:
> If it is really foreseeable that this function gets removed in the near future then we can merge this as is. But if not then we should improve this and not beautify the html rendering. Also the issue does not mention anything about the bullet points.
>

It will surely not be updated in _near_ future, as the feature seems to be added for version 6.0.0 of this npm package. 
Still you are right it is not part of the issue, but it seems to fall into the same category. The behavior without this custom renderer is that the bullet points are rendered on the left side so it is rather a beauty-related issue.

> I dont think this feature is that important that it justifies to use internal API of an alpha npm module.
the function is also used in their demo on how to implement a renderer see [here](https://github.com/archriss/react-native-render-html/blob/9f7168be5ceb1cf4de6e51d3a93bd9cca07e5c3d/Demo/src/snippets.js#L273)

Personally I don't mind leaving this part out.

Reply:
Honestly I don't see why we should remove this part again as it fixes a RTL issue just for the sake of not using internal API. I don't like it either here, but I don't think we should spent more time here.